message_ix_models/model/water/data/water_for_ppl.py

```diff
@@ -85,26 +96,36 @@ def shares(
     corresponding cooling fraction
     """
     for col in search_cols_cooling_fraction:
-        # MAPPING ISOCODE to region name, assume one country only
         col2 = context.map_ISO_c[col] if context.type_reg == "country" else col
+
+        # Filter the cooling fraction
         cooling_fraction = hold_df[
             (hold_df["node_loc"] == col2)
             & (hold_df["technology_name"] == x["technology"])
         ]["cooling_fraction"]
-        x[col] = x[col] * cooling_fraction
 
-    results: list[Any] = []
+        # Log unmatched rows
+        if cooling_fraction.empty:
+            log.warning(
+                f"No cooling_fraction found for node_loc: {col2}, technology: {x['technology']}"
+            )
+            cooling_fraction = pd.Series([0])
+
+        # Ensure the Series is not empty before accessing its first element
+        if not cooling_fraction.empty:
+            x[col] = x[col] * cooling_fraction.iloc[0]
+        else:
+            # You could choose to leave the value unchanged or apply a default
+            x[col] = x[col] * 0  # Default to 0 if cooling_fraction is empty
```

Comment:
```suggestion
        # You could choose to leave the value unchanged or apply a default
        # Default to 0 if cooling_fraction is empty
        x[col] = x[col] * cooling_fraction.iloc[0] if not cooling_fraction.empty else x[col] * 0  
```
This again needs ruff, but avoids tests and complexity.